### PR TITLE
OCPBUGS-11291: skip when ListImage with filter image hasn't done pulling

### PR DESCRIPTION
**Pull request by QiWang19**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Apply a previous fix https://github.com/cri-o/cri-o/pull/4726 to ListImage with `filter`.
If the file not found error was caused by image not done pulling when ListImages() with filter, skip the image and the error.

Previous bug https://bugzilla.redhat.com/show_bug.cgi?id=1942608  appeared in https://issues.redhat.com/browse/OCPBUGS-11291. 
The log hyperkube ListImages with filters. `"ListImages with filter from image service failed" err="rpc error: code = Unknown desc = error locating item named \"manifest\" for image with ID \"9e11cc7364c9791ebf68949c852bf121cb4ae5829c925a1da6546be6f63b3536\": file does not exist" filter="nil"`
#### Which issue(s) this PR fixes:
https://issues.redhat.com/browse/OCPBUGS-11291

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: internal/storage/image.go [hunk 1]
```diff
@@ -374,6 +372,17 @@ func (svc *imageService) ListImages(systemContext *types.SystemContext, filter s
 	return results, nil
 }
 
+func imageDonePulling(image *storage.Image) bool {
+	donePulling := true
+	for _, name := range image.Names {
+		if _, ok := ImageBeingPulled.Load(name); ok {
+			donePulling = false
```
Comment (haircommander) [inline internal/storage/image.go:379]: we can just `return false` here, and `return true` below, no need for donePulling if we separate the functions

Comment (QiWang19) [reply internal/storage/image.go]: thanks. fixed

**Comment by QiWang19** [comment]
/hold
I am not sure this fix makes sense.

**Comment by QiWang19** [comment]
I don't think this PR makes sense for the bug.

**Pull request closed without merging**